app/content/profile/index.js

```diff
@@ -65,30 +74,178 @@ function Index(props) {
+      setSnackbarVisible(true); // Show success snackbar
+    }
+
+    if (email) {
+      // TODO: Decide whether we want to allow a user update their email
+    }
+
+    if (passwordInputVisible && (currentPassword || newPassword)) {
+      if (!currentPassword || !newPassword) {
+        showDialog("Error", "Please fill out all the fields");
+      } else {
+        // attempt updating the password
+        try {
+          // re-authenticate the user and check if the provided current password is valid
+          const userCredential = await signInWithEmailAndPassword(
+            auth,
+            userEmail,
+            currentPassword,
+          );
+
+          // once re-authenticated, update the password
+          updatePassword(userCredential.user, newPassword)
+            .then(() => {
+              // Update successful
+              setCurrentPassword(""); // Clear password fields
+              setNewPassword("");
+              bottomSheetModalRef.current.close();
+              setSnackbarMessage("Password updated successfully");
+              setSnackbarVisible(true); // Show success snackbar
+            })
+            .catch((error) => {
+              // Update failed
+              console.log("password update error:", error.message);
+              showDialog(
+                "New password is too short",
+                "Provided new password must be at least 6 characters long!",
+              );
+            });
+        } catch (e) {
+          showDialog("Error", e.message);
+          console.log(e.message);
+        }
+      }
+    }
+  };
+
   const uniqueDrills = getUnique(attempts, "did");
 
   return (
     <PaperProvider>
+      <Snackbar
+        visible={snackbarVisible}
+        onDismiss={() => setSnackbarVisible(false)}
+        duration={4000} // Duration in milliseconds for how long the snackbar is shown
+      >
+        {snackbarMessage}
+      </Snackbar>
+
+      <Portal>
+        <Dialog
+          visible={dialogVisible}
+          onDismiss={() => setDialogVisible(false)}
+        >
+          <Dialog.Title>{dialogTitle}</Dialog.Title>
+          <Dialog.Content>
+            <Paragraph>{dialogMessage}</Paragraph>
+          </Dialog.Content>
+          <Dialog.Actions>
+            <Button onPress={() => setDialogVisible(false)}>OK</Button>
+          </Dialog.Actions>
+        </Dialog>
+      </Portal>
+
       <SafeAreaView style={{ flex: 1 }}>
         <Appbar.Header statusBarHeight={0} style={{ backgroundColor: "FFF" }}>
           <Appbar.Content title={"Personal Profile"} />
           <Appbar.Action
             icon="cog"
             color={"#F24E1E"}
-            onPress={handlePresentModalPress}
+            onPress={() => bottomSheetModalRef.current?.present()}
             style={{ marginRight: 7 }}
           />
         </Appbar.Header>
+
         <BottomSheetModalProvider>
           <ScrollView contentContainerStyle={styles.scrollViewContent}>
```

Comment:
Since keyboard inputs are getting involved, I recommend using KeyboardAwareScrollView library to avoid some iOS or Android specific bugs.

`import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view";`
(it's already in package.json, and the props passed to it are same as regular scrollview)

Example on Android where it's hard to see the confirm password input field (when updating password)
![image](https://github.com/Golf-Drill-Challenge-App/Golf-App/assets/82061589/e927ef9c-c7e2-4d32-bcbc-3cbb4310978e)